### PR TITLE
re: Move types only used by `PeerStore` into the same file

### PR DESCRIPTION
**Pull request by pmnoxx**
`KnownPeerState` and `KnownPeerStatus` are only used by `PeerStore`.
It makes sense to move them from `near-network-primitives` inside `peer_store.rs`.

### CODE REVIEW

File: chain/network-primitives/src/types.rs [hunk 1]
```diff
@@ -648,60 +645,6 @@ impl FromStr for PatternAddr {
     }
 }
```
Comment (mm-near) [inline chain/network-primitives/src/types.rs:647]: I'd normally agree - but at the same time, these are BorchSerialized - which means that we persist them somewhere etc.

Wondering if maybe we should keep such kinds of types in network-primitives.. (but that requires a little bit more discussion)

Comment (pmnoxx) [reply chain/network-primitives/src/types.rs:647]: @matklad @mm-near Those types shouldn't be visible to the outside world. As, they are used only by `near-network`.

Proposals:
A) Move to `peer_store.rs`
B) That being said, having a separate `db_types.rs`, which is private inside `near-network` would be an alternative option. That way, we can more easily control any changes.
C) Keep as it is.

It seems to me that `A,B > C`. I'm fine with either `A` or `B`.

File: chain/network/src/peer_manager/db_types.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    pub(crate) fn new(peer_info: PeerInfo) -> Self {
+        KnownPeerState {
+            peer_info,
+            status: KnownPeerStatus::Unknown,
+            first_seen: to_timestamp(Clock::utc()),
+            last_seen: to_timestamp(Clock::utc()),
+        }
+    }
+
+    pub(crate) fn last_seen(&self) -> DateTime<Utc> {
```
Comment (matklad) [inline chain/network/src/peer_manager/db_types.rs:44]: would be nice to get rid of chrono dependency here while we are at it, and if that's not too much bigger diff.

File: chain/network/src/peer_manager/db_types.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use chrono::{DateTime, Utc};
+use near_network_primitives::types::{PeerInfo, ReasonForBan};
+use near_primitives::time::Clock;
+use near_primitives::utils::{from_timestamp, to_timestamp};
+
+/// Status of the known peers.
+#[derive(BorshSerialize, BorshDeserialize, Eq, PartialEq, Debug, Clone)]
+pub enum KnownPeerStatus {
```
Comment (matklad) [inline chain/network/src/peer_manager/db_types.rs:9]: Can we `pub(crate)` it to make it obvious that this isn't used outside?

**Review by matklad — APPROVED** [review]

File: chain/network/src/peer_manager/db_types.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network/src/peer_manager/db_types.rs:44]: @matklad See https://github.com/near/nearcore/pull/5556#pullrequestreview-831602436
It has been stuck in review queue for 2 weeks.

File: chain/network/src/peer_manager/db_types.rs (hunk 3 above)
Comment (pmnoxx) [reply chain/network/src/peer_manager/db_types.rs:10]: @matklad It's needed.
I added the following.
```
/// `pub` is used by `test-utils` in `iter_peers_from_store`
```

Maybe, we should rewrite `iter_peers_from_store` not to export private information from this module.

**Comment by pmnoxx** [comment]
```
@mm-near: chain/network/ chain/network-primitives/
```
@mm-near Can you take a look?

**Comment by pmnoxx** [comment]
```
@mm-near: chain/network/ chain/network-primitives/
```
@mm-near Can you take a look?

**Pull request closed without merging**